gradio/blocks.py

```diff
@@ -726,15 +733,14 @@ def iterate_over_children(children_list):
                 # targets field
                 _targets = dependency.pop("targets")
                 trigger = dependency.pop("trigger", None)
-                targets = [
-                    getattr(
-                        original_mapping[
-                            target if isinstance(target, int) else target[0]
-                        ],
-                        trigger if isinstance(target, int) else target[1],
-                    )
-                    for target in _targets
-                ]
+                is_then_event = False
+
+                # This assumes that you cannot combine multiple .then() events in a single
+                # gr.on() event, which is true for now. If this changes, we will need to
+                # update this code.
+                if not isinstance(_targets[0], int) and _targets[0][1] == "then":
```

Comment:
Let's add a test that would break if this assumption changes!

Reply:
Good idea

Reply:
I added an assertion that would tell us if this assumption is violated